packages/server/src/auth/auth.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable } from '@nestjs/common';
+import * as bcrypt from 'bcrypt';
+import { PrismaService } from '../prisma/prisma.service';
+import { UsersService } from '../users/users.service';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class AuthService {
+  constructor(private usersService: UsersService, private prisma: PrismaService, private jwtService: JwtService) {}
+
+  async validateUsername(username: string, password: string): Promise<any> {
+    const user = await this.usersService.findOneUsername(username);
+
+    if(user && await bcrypt.compare(password, user.password)) {
```

Comment:
When the user is not found, it will throw a NotFoundError instead of returning a null.

Reply:
Do we want to throw an error? I tested logging in with a username that's not in the db, and it returns a 401, which is correct.

Reply:
Asked Greg about how should find user functions response when the user does not exist. It should throw a NotFoundError. Not sure about how AuthService should deal with the Error threw back. @gregfrasco So should we catch the Error in the AuthService do something about it?

Reply:
If user does not exist, generally should log, but not return specific indication that the user does not exist to the client. That would give away information about what information exists in the db. Instead client should indicate to the user that the credentials are not correct.

Returning null here should be fine. The main consideration should be on how we want to handle logging.